tests/publiccloud/upload_image.pm

```diff
@@ -32,6 +32,12 @@ sub run {
         return;
     }
 
+    # Check if remote file exists
+    my $res = script_output("wget -q --spider -S $img_url");
+    if ($res =~ m@HTTP/1.1 404 Not Found@) {
```

Comment:
Also working by excluding version and slash: m/HTTP.*404 Not Found/,
but ok as it is now, LGTM.

Reply:
Yeah, I thought about that. I will update the regex. Thanks!